app/assets/javascripts/pretty-text/engines/discourse-markdown/code.js

```diff
@@ -1,35 +1,72 @@
 // we need a custom renderer for code blocks cause we have a slightly non compliant
 // format with special handling for text and so on
-
 const TEXT_CODE_CLASSES = ["text", "pre", "plain"];
 
-function render(tokens, idx, options, env, slf, md) {
-  let token = tokens[idx],
-    info = token.info ? md.utils.unescapeAll(token.info) : "",
-    langName = md.options.discourse.defaultCodeLang,
-    className,
-    escapedContent = md.utils.escapeHtml(token.content);
-
-  if (info) {
-    // strip off any additional languages
-    info = info.trim().split(/\s+/g)[0];
+function extractTokenInfo(info, md) {
+  if (!info) {
+    return;
+  }
+
+  info = md.utils.unescapeAll(info.replace(/[^\x00-\x7F]/g, "")).trim();
+
+  const matches = info.match(/(^\s*\S*)\s*(.*)/i);
+  if (!matches) {
+    return;
+  }
+
+  // ensure the token has only valid chars
+  if (!/^\w*$/i.test(matches[1])) {
```

Comment:
The 'token' is the name of the language, right? 

I think we might need to be more permissive here for things like `c++` and `obj-c`? ([language list here](https://highlightjs.readthedocs.io/en/latest/supported-languages.html))

Reply:
Hmmm yes, I see good point. I have been extreme like this due to the common mark spec, I will try to handle both cases

Reply:
https://github.com/discourse/discourse/pull/15743/commits/44bd3e06b8b9b60e11761d57213de6c74ee19877

Reply:
[FIX: more accurate tag validity check](https://github.com/discourse/discourse/pull/15743/commits/f97a9a115cdf32e9508db5c4c0cbc5a47a6064cf)